Mathlib/Algebra/GroupWithZero/Divisibility.lean

```diff
@@ -9,14 +9,16 @@ Neil Strickland, Aaron Anderson
 ! Please do not edit these lines, except to modify the commit id
 ! if you have ported upstream changes.
 -/
-import Mathlib.Algebra.GroupWithZero.Basic
+import Mathlib.Algebra.GroupWithZero.Units.Basic
 import Mathlib.Algebra.Divisibility.Units
 
 /-!
 # Divisibility in groups with zero.
 
 Lemmas about divisibility in groups and monoids with zero.
 
+We also define `Ring.divide`, a globally defined function on any ring
```

Comment:
Can you please elaborate on how/where you want to use this function? Would a `DivInvMonoid` work in your setting?

Reply:
I'm looking at polynomial rings over (say) a commutative ring, which a priori don't have division/inversion but the notion of "cancelling out factors formally" still makes sense.

Reply:
Shouldn't we treat division in polynomial rings the same way we treat it in `Nat` and `Int`? So we can just use the ordinary division symbol for some "truncated division" operation.

Reply:
Yes I think we should, but I don't know how to best introduce this. Define `Ring.divide` as I have and then add notation?

Reply:
Isn't this what `rem` and `quot` are for? I don't remember what typeclass they are from...

Reply:
There are `EuclideanDomain.quotient` and `EuclideanDomain.remainder` but my polynomial rings are not Euclidean.